tools/cloud-build/daily-tests/ansible_playbooks/tasks/create_deployment_directory.yml

```diff
@@ -0,0 +1,52 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+---
+- name: Assert variables are defined
+  ansible.builtin.assert:
+    that:
+    - blueprint_yaml is defined
+    - project is defined
+    - workspace is defined
+    - deployment_name is defined
+    - test_name is defined
+
+- name: Create cli flag for extra deployment variables
+  ansible.builtin.set_fact:
+    deployment_vars_str: "--vars {{ cli_deployment_vars.items() | map('join', '=') | join(',') }}"
+  when: cli_deployment_vars is defined
+
+- name: Create Blueprint
+  ansible.builtin.command: |
+      ./ghpc create -l ERROR "{{ blueprint_yaml }}" \
+      --backend-config bucket=daily-tests-tf-state \
+      {{ deployment_vars_str if deployment_vars_str is defined else '' }} \
+      --vars project_id={{ project }} \
+      --vars deployment_name={{ deployment_name }}
```

Comment:
Just to state, this positioning of project_id and deployment_name should "win" over values in the mapping. If this is desirable (it probably is), please close this comment.

Reply:
Good point. I think if test explicitly sets that it should take precedence. I switched the order to put `deployment_vars_str` last. I also tested to make sure having final line as empty string does not cause failure of command.